internal/git/git.go

```diff
@@ -147,15 +147,12 @@ func CurrentUpstreamBranch() (string, error) {
 
 // UpstreamBranch returns the upstream of the specified branch
 func UpstreamBranch(branch string) (string, error) {
-	cmd := New("rev-parse", "--abbrev-ref", branch+"@{upstream}")
-	cmd.Stdout = nil
-	cmd.Stderr = nil
-	ref, err := cmd.Output()
+	upstreamBranch, err := gitconfig.Local("branch." + branch + ".merge")
```

Comment:
Sorry, but I didn't get the reason for using `branch.<name>.merge` instead of `git rev-parse --abbrev-ref <name>@{usptream}`. At the same time, how this change makes the code "less restrictive" wrt local remote naming?
Could you share a case or workflow where your change has clear advantage?

Reply:
The issue arises when having a remote with at least a '/' in its name. Say we have a remote called `github/username` and a branch named `branch/name`, `git rev-parse --abbrev-ref branch/name@{u}` will return `github/username/branch/name`. 

`UpstreamBranch` only returns the remote branch name, and was splitting the remote name from the branch name based on the first slash, using `strings.SplitN`. That is not valid as remotes can have slashes in their names (and more than one).

We want to fix the logic of removing the remote name from the value returned by `git rev-parse`, however finding this information isn't easy (expect if we loop over all the remote names). But looking at `git rev-parse` documentation, it says the `@{upstream}` value is constructed using `branch.<name>.remote` and `branch.<name>.merge` (meaning we were using `@{upstream}` which was adding the remote name to the branch one, to then remove the remote name part).

Given that, a clean solution is to directly use `branch.<name>.merge` as it is what we want: the remote branch name, without the remote name included. The only thing we need to take care of is that string is prefixed with 'refs/heads/'.

Example:
```
$ git remote
gitlab/username
gitlab/project
$ git rev-parse --abbrev-ref HEAD
my/branch
$ git rev-parse --abbrev-ref @{u}
gitlab/username/my/branch
$ lab mr create gitlab/project
mr_create.go:128: aborting MR, source branch username/my/branch not present on remote gitlab/username.
```

(I just realized we might want to check the string is at least 11 chars long before removing the first 11 chars).

Reply:
> (I just realized we might want to check the string is at least 11 chars long before removing the first 11 chars).

Or use `strings.TrimPrefix(upstreamBranch, "refs/heads/")`

Reply:
Right. I always have the impression using those functions when the argument can't be changed at runtime is not ideal. But we're talking about a single call, so I can use `strings.TrimPrefix` if necessary.

Reply:
Ah ok. I see the issue now. Thanks for the clarification!!
And I agree with @fmuellner here.. just use `strings.TrimPrefix` here.

Reply:
All right, I'll use `strings.TrimPrefix` :)